docs/DafnyRef/UserGuide.md

```diff
@@ -315,7 +315,260 @@ This list is not exhaustive but can definitely be useful to provide the next ste
   <br><br>`assert forall i | i == m :: P(m);` |  `assert P(m);`<br>`assert forall i | i == m :: P(i);`
   `method m(i) returns (j: T)`<br>&nbsp;&nbsp;`  requires A(i)`<br>&nbsp;&nbsp;`  ensures B(i, j)`<br>`{`<br>&nbsp;&nbsp;`  ...`<br>`}`<br><br>`method n() {`<br>&nbsp;&nbsp;`  ...`<br><br><br>&nbsp;&nbsp;`  var x := m(a);`<br>&nbsp;&nbsp;`  assert P(x);` | `method m(i) returns (j: T)`<br>&nbsp;&nbsp;`  requires A(i)`<br>&nbsp;&nbsp;`  ensures B(i, j)`<br>`{`<br>&nbsp;&nbsp;`  ...`<br>`}`<br><br>`method n() {`<br>&nbsp;&nbsp;`  ...`<br>&nbsp;&nbsp;`  assert A(k);`<br>&nbsp;&nbsp;`  assert forall x :: B(k, x) ==> P(x);`<br>&nbsp;&nbsp;`  var x := m(k);`<br>&nbsp;&nbsp;`  assert P(x);`
 
-### 24.8.2. Assertion batches {#sec-assertion-batches}
+### 24.8.2. Verification debugging when verification is slow {#sec-verification-debugging-slow}
+
+In this section, we describe methodologies to apply in the case when verification is slower than expected, does not terminate, or timeouts.
+
+#### 24.8.2.1. `assume false;` {#sec-assume-false}
+
+Assuming `false` is a powerful way to short-circuit the verifier and stop verification at a given point[^explainer-assume-false], and since the final compilation steps do not accept this command, it is safe to use it during development.
```

Comment:
Just FYI, I'm a little wary of documenting this so strongly since its effectiveness is dependent on undefined behavior of the verifier: https://github.com/dafny-lang/dafny/issues/1441

This looks like great content that seems to work well now, just might need a caveat like what I discuss in that issue.

Reply:
I'm bothered by the fact that the compiler rejects `assume false` — it seems quite arbitrary.  I'm not sure it's worth documenting here.

Reply:
> Just FYI, I'm a little wary of documenting this so strongly since its effectiveness is dependent on undefined behavior of the verifier: https://github.com/dafny-lang/dafny/issues/1441

Interesting. What about making "assume false; X" officially recognized by the Dafny verifier (like what you suggested `unverified { X }` so that it will not emit any assert in that block?
Since it seems this works well, I'll reword to ensure I describe it as "experimental" or "know-how".

> I'm bothered by the fact that the compiler rejects assume false — it seems quite arbitrary. I'm not sure it's worth documenting here.

I think it's for the same legitimate reason as rejecting functions or methods without bodies if they are not `{:extern}`. There is nothing to back an `assume` (not just `assume false;`, but every `assume`)

Reply:
Here is my new reformulation that is less emphasizing:

"Assuming `false` is an empirical way to short-circuit the verifier and usually stop verification at a given point"